capellambse_context_diagrams/context.py

```diff
@@ -343,18 +343,14 @@ def render(self, fmt: str | None, /, **params) -> t.Any:
 
     def _create_diagram(self, params: dict[str, t.Any]) -> cdiagram.Diagram:
         transparent_background = params.pop("transparent_background", False)
-        self.display_parent_relation = params.pop(
-            "display_parent_relation", self.display_parent_relation
-        )
-        self.display_derived_interfaces = params.pop(
-            "display_derived_interfaces", self.display_derived_interfaces
-        )
-        self.display_symbols_as_boxes = params.pop(
-            "display_symbols_as_boxes", self.display_symbols_as_boxes
-        )
-        self.slim_center_box = params.pop(
-            "slim_center_box", self.slim_center_box
-        )
+        for param_name in [
+            "display_parent_relation",
+            "display_derived_interfaces",
+            "display_symbols_as_boxes",
+            "slim_center_box",
+        ]:
+            default = getattr(self, param_name)
+            setattr(self, param_name, params.pop(param_name, default))
```

Comment:
This feels a bit backward to me.

```suggestion
            override = params.pop(param_name, None)
            if override is not None:
                setattr(self, param_name, override)
```

IMO an even better solution to this though would be to keep all the user-provided options in `params` instead of extracting some of them as instance variables. But this would affect part of the [documented public API](https://dsd-dbs.github.io/capellambse-context-diagrams/extras/styling/#no-symbol-rendering), and I'm not entirely sure how best to cope with that. For the sake of un-breaking the rendering cache (which is currently broken if you set instance variables instead of using `params`), I'd like to see everything migrated into `params` sooner or later.

Reply:
> This feels a bit backward to me.
> 
> IMO an even better solution to this though would be to keep all the user-provided options in `params` instead of extracting some of them as instance variables. But this would affect part of the [documented public API](https://dsd-dbs.github.io/capellambse-context-diagrams/extras/styling/#no-symbol-rendering), and I'm not entirely sure how best to cope with that. For the sake of un-breaking the rendering cache (which is currently broken if you set instance variables instead of using `params`), I'd like to see everything migrated into `params` sooner or later.

@Wuestengecko is very right here. No need to get the default value, just override if there is a match from params:
```suggestion
            if override := params.pop(param_name, None):
                setattr(self, param_name, override)
```

Reply:
Be aware that these two suggestions aren't exactly semantically equivalent. `if override := params.pop(param_name, None):` does not branch into the `setattr` call if the `override` is *any* falsey value, not only None - like 0, False, an empty string/list/tuple etc. To be specific, it wouldn't be possible to set `slim_center_box` to False anymore.